pkg/driver/qemu/qemu_driver.go

```diff
@@ -27,36 +27,51 @@ import (
 	"github.com/sirupsen/logrus"
 
 	"github.com/lima-vm/lima/pkg/driver"
+	"github.com/lima-vm/lima/pkg/driver/qemu/entitlementutil"
 	"github.com/lima-vm/lima/pkg/executil"
 	"github.com/lima-vm/lima/pkg/limayaml"
 	"github.com/lima-vm/lima/pkg/networks/usernet"
 	"github.com/lima-vm/lima/pkg/osutil"
-	"github.com/lima-vm/lima/pkg/qemu/entitlementutil"
 	"github.com/lima-vm/lima/pkg/store"
 	"github.com/lima-vm/lima/pkg/store/filenames"
 )
 
 type LimaQemuDriver struct {
-	*driver.BaseDriver
+	Instance     *store.Instance
+	SSHLocalPort int
+	VSockPort    int
+	VirtioPort   string
+
+	DriverType driver.DriverType
```

Comment:
It feels like a failure of the abstraction if the driver needs to know if it is an internal or external driver. I was hoping that this could be entirely handled by the remoting layer.

Reply:
GuestAgentConn needs to know this, at least for now

Reply:
One way to deal with this would be to have both a `GuestAgentConn()` and `ExternalGuestAgentConn()` method on the driver, and the `driverserver.GuestAgentConn()` method in `pkg/driver/external/server/methods.go` would call `s.driver.ExternalGuestAgentConn()` instead of the internal version.

Maybe call it `GuestAgentConnExternal()` so both have the same prefix, idk.

Reply:
> change the method signature to `GuestAgentConn() (net.Conn, type string)`

That would be even better, if you can make it work. Because then the proxying is not part of the driver, but of the server code, where it really belongs.

I just think that the driver should be completely ignorant of whether it is running internal or external.